### PR TITLE
Support wellknown

### PR DESCRIPTION
**Pull request by bkaney**
WIP for smart and oidc wellknown

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -201,11 +202,78 @@ class Client {
...
+    const errors = [];
+
+    return new Promise((resolve, reject) => {
+      function handleError(error) {
+        if (errors.push(error) === queue.numJobs) {
+          reject(new Error(errors.map((e) => e.message).join('; ')));
+        }
+      }
+
+      const wellKnownSmartUrl = `${this.baseUrl.replace(/\/*$/, '/')}.well-known/smart-configuration`;
```
Comment (jpowers) [inline lib/client.js:254]: It appears the standard `.well-known` URI for OIDC is `/.well-known/openid-configuration`. That's also what SmileCDR uses. That's different than what the FHIR standard is for SMART.  ¯\_(ツ)_/

Comment (jpowers) [reply lib/client.js:254]: Oh I see you cover that below. Nice.

**Comment by jpowers** [comment]
You should add `application/json` to the `Accept` header.
`Accept: application/fhir+json, application/json`
Or just use `application/json`
Also change the default to `application/fhir+json`.

**Comment by bkaney** [comment]
> You should add `application/json` to the `Accept` header.
> `Accept: application/fhir+json, application/json`
> Or just use `application/json`
> Also change the default to `application/fhir+json`.

Yep. I added this in as well. @jpowers can you take a look and test it out? I am going to do a few more real-world tests myself, but I think this is getting very close.

File: lib/smart.js [hunk 2]
```diff
@@ -50,6 +50,21 @@ function authFromCapability(capabilityStatement) {
   }
 }
 
+/* eslint-disable camelcase */
+function authFromWellKnown(wellKnown) {
+  const {
+    authorization_endpoint,
+    token_endpoint,
+    registeration_endpoint,
```
Comment (vstrateman) [inline lib/smart.js:58]: registration is spelled incorrectly here

**Comment by jpowers** [comment]
Looks good.  I tested and everything works on my end

**Pull request merged**